loadtest/control/simulcontroller/actions.go

```diff
@@ -556,6 +559,78 @@ func deletePost(u user.User) control.UserActionResponse {
 	return control.UserActionResponse{Info: fmt.Sprintf("post deleted, id %v", post.Id)}
 }
 
+func (c *SimulController) createSidebarCategory(u user.User) control.UserActionResponse {
+	team, err := u.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	category := &model.SidebarCategoryWithChannels{
+		SidebarCategory: model.SidebarCategory{
+			UserId:      u.Store().Id(),
+			TeamId:      team.Id,
+			DisplayName: "category" + control.PickRandomWord(),
+		},
+	}
+
+	sidebarCategory, err := u.CreateSidebarCategory(u.Store().Id(), team.Id, category)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.UserActionResponse{Info: fmt.Sprintf("created sidebar category, id %s", sidebarCategory.Id)}
+}
+
+func (c *SimulController) updateSidebarCategory(u user.User) control.UserActionResponse {
+	team, err := u.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if team == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	cat1, err := u.Store().RandomCategory(team.Id)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	cat2, err := u.Store().RandomCategory(team.Id)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	// Not repeatedly looping until we get a different category because there have been edge-cases before
+	// ending in infinite loop.s
+	if cat1.Id == cat2.Id {
+		return control.UserActionResponse{Info: "same categories returned. Skipping."}
+	}
+	if len(cat1.Channels) <= 1 {
+		return control.UserActionResponse{Info: "Not enough categories to remove. Skipping."}
+	}
+
+	// We pick a random channel from first category and move to second category.
+	channelToMove := control.PickRandomString(cat1.Channels)
+
+	// Find index
+	i := 0
+	for i = range cat1.Channels {
+		if cat1.Channels[i] == channelToMove {
+			break
+		}
+	}
+	// Move from the first, and add to second.
+	cat1.Channels = append(cat1.Channels[:i], cat1.Channels[i+1:]...)
+	cat2.Channels = append(cat2.Channels, channelToMove)
```

Comment:
I am guessing this is safe for now but given we don't really copy I am thinking it could become a problem if we were to do anything on them from another goroutine (e.g. the websocket side).

Reply:
yeah good point